apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -88,6 +89,8 @@ def __init__(
         reindex_disable_on_axis: Optional[Union[int, Sequence[int]]] = None,
         eager: bool = True,
         eager_iterator_pool: Optional[ThreadPoolExecutor] = None,
+        compress: bool = True,
```

Comment:
Moving the `compress` flag from BlockwiseScipyReadIter to the base class seems incorrect.  The flag only has meaning in the context of the scipy iterators (i.e., it is a flag which controls the choice of a compressed sparse matrix vs. a COO matrix).